src/spi.rs

```diff
@@ -856,10 +908,14 @@ where
         operations: &mut [(&mut [u8], &[u8])],
     ) -> Result<(), EspError> {
         self.run(
-            &mut operations
+            self.hardware_cs_ctl(
+                operations
+                    .iter_mut()
+                    .map(|(read_slice, write_slice)| Operation::Transfer(read_slice, write_slice)),
+            )?,
+            operations
                 .iter_mut()
                 .map(|(read_slice, write_slice)| Operation::Transfer(read_slice, write_slice)),
```

Comment:
I don't have an idea of how yet but it'd be nice if this code could be de-duplicated somehow.
Particularly the `operations.iter_mut().map(|(read_slice, write_slice)| Operation::Transfer(read_slice, write_slice))` bit.
Maybe some kinda `IntoIterator` kinda deal. I'll think about this some more.

Reply:
Actually, is there a reason to keep `transfer_transaction` around? `e-hal` doesn't need it anymore and users can easily use the `transaction` API.

Reply:
I tried your into iterator idea tree times and failed each time as I could not figure out the lifetimes of the `impl Iterator` thing returned by it. Not even sure that impl Trait is expressive enough to capture these lifetimes by the way, but if you feel like it, give it a whirl yourself.

Reply:
> Actually, is there a reason to keep `transfer_transaction` around? `e-hal` doesn't need it anymore and users can easily use the `transaction` API.

I'll check, that's a good point.